hazelcast/src/test/java/com/hazelcast/client/map/impl/nearcache/ClientMapNearCacheInvalidationTest.java

```diff
@@ -158,6 +169,216 @@ public void testMapClear_shouldClearNearCaches_onOwnerAndBackupNodes() {
         assertNearCacheSizeEventually(clientMap, 0);
     }
 
+    @Test
+    public void testMapSubmitToKey_withReadOnlyProcessor_noInvalidations() {
```

Comment:
The 

```java
Config config = getConfig();
        configureBatching(config, 10, 1);

        ClientConfig clientConfig = getClientConfig(mapName);

        HazelcastInstance server = factory.newHazelcastInstance(config);
        factory.newHazelcastInstance(config);

        HazelcastInstance client = factory.newHazelcastClient(clientConfig);
        makeSureConnectedToServers(client, 2);

        IMap<Integer, Integer> serverMap = server.getMap(mapName);
        IMap<Integer, Integer> clientMap = client.getMap(mapName);

        int size = 1000;

        // fill serverMap
        for (int i = 0; i < size; i++) {
            serverMap.put(i, i);
        }

        populateNearCache(clientMap, size);
        NearCacheStats nearCachestats = ((NearCachedClientMapProxy) clientMap).getNearCache().getNearCacheStats();
        long invalidationsBefore = nearCachestats.getInvalidations();
```

parts are the same in 4 tests starting from this, you should introduce a method I think

Reply:
(similar for the NearCacheTest file)

Reply:
Resolving because in slack we agreed this is not a must